airbyte-cdk/python/airbyte_cdk/sources/declarative/partition_routers/substream_partition_router.py

```diff
@@ -127,12 +130,14 @@ def stream_slices(self) -> Iterable[StreamSlice]:
                 parent_stream = parent_stream_config.stream
                 parent_field = parent_stream_config.parent_key.eval(self.config)  # type: ignore # parent_key is always casted to an interpolated string
                 partition_field = parent_stream_config.partition_field.eval(self.config)  # type: ignore # partition_field is always casted to an interpolated string
+                incremental_dependency = parent_stream_config.incremental_dependency
                 for parent_stream_slice in parent_stream.stream_slices(
                     sync_mode=SyncMode.full_refresh, cursor_field=None, stream_state=None
                 ):
-                    empty_parent_slice = True
                     parent_partition = parent_stream_slice.partition if parent_stream_slice else {}
 
+                    # we need to read all records for slice to update the parent stream cursor
+                    stream_slices_for_parent = []
                     for parent_record in parent_stream.read_records(
```

Comment:
let's add a comment here explaining that the sync mode and the stream state passed are irrelevant since we set the initial state from `set_parent_state`

Reply:
Fixed